test/test_profiler.py

```diff
@@ -502,6 +502,29 @@ def test_flops(self):
             sort_by="self_cuda_time_total", row_limit=-1)
         self.assertIn("Total MFLOPs", profiler_output)
 
+    def test_custom_function_shapes(self):
+        class DummyFunction(torch.autograd.Function):
+            @staticmethod
+            def forward(ctx, tensor, constant):
+                ctx.constant = constant
+                return tensor * constant
+
+            @staticmethod
+            def backward(ctx, grad_output):
+                return grad_output * ctx.constant, None
+
+        func = DummyFunction.apply
+
+        with _profile(with_stack=True, use_kineto=kineto_available(), record_shapes=True) as p:
+            u = torch.randn(3, 4, 5, requires_grad=True)
+            v = func(u, 2)
+            w = v.sum()
+            w.backward()
+
+        for e in p.function_events:
+            if "DummyFunction" in e.name:
```

Comment:
@moderato Probably should make sure `DummyFunction` is found in `p.function_events`, e.g., set a flag `found_func = True` inside the `if`, then `assertTrue` after the `for` loop.

Reply:
Sure, done.